person.rb

```diff
@@ -1,11 +1,21 @@
-require './nameable'
-# Public: Person class used for creating various person objects.
+# Public: Nameable class is the base component of the Person class.
+#
+# All methods are class methods and should be called on the nameable class.
+class Nameable
+  # Public: Returns NotImplementedError message of type String.
+  def correct_name
+    raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
+  end
+end
```

Comment:
- [ ] (OPTIONAL)  Great job so far 👍 It would have been better to put the `Nameable` class in its own ruby file so as to keep your work neat. The same thing goes for the `TrimmerDecorator` and `CapitalizeDecorator` classes 💯

Reply:
Thank you. I will do that now.